File: apiaryFrontend/src/app/models/note.ts
export class Note {
    id!: number;
    date!: Date;
    text!: string;
    hiveId!: number;
    honey!: number;
    syroup!: number;

    constructor(id: number, date: Date, text: string, hiveId: number, honey: number, syroup: number) {
        this.id = id;
        this.date = date;
        this.text = text;
        this.hiveId = hiveId;
        this.honey = honey;
        this.syroup = syroup;
    }

}
//Work is note without id
export class Work{
    date!: Date;
    name!: string;
    text!: string;
    hiveId!: number;
    honey!: number;
    syrup!: number;

    constructor(date: Date, name:string, text: string, hiveId: number, honey: number, syrup: number) {
        this.date = date;
        this.name = name;
        this.text = text;
        this.hiveId = hiveId;
        this.honey = honey;
        this.syrup = syrup;
    }
}

File: apiary/src/main/java/com/apiary/note/Note.java
package com.apiary.note;

import com.apiary.hive.Hive;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;

@Data
@Entity
public class Note {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Date date;
    private String text;
    private int honey;
    private int syrup;

    @ManyToOne
    @JoinColumn(name = "hive_id")
    private Hive hive;

    // Getters and Setters
}

